test/functional/tests/volumes/test_volume_partitions.py

```diff
@@ -0,0 +1,192 @@
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/cas"
+core_number = 16
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.parametrize("partition_table", PartitionTable)
+@pytest.mark.parametrize("filesystem", Filesystem)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_partition_preserve_cas(partition_table, filesystem, cache_mode):
+    """
+        title: Volume test for preserving partition table from core device.
+        description: |
+          Validation of the ability of CAS to preserve partition table on core device
+          after running cache.
+        pass_criteria:
+          - Md5 sums shall on partitions be identical before and after running cache.
+    """
+    with TestRun.step(f"Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_size = []
+        for i in range(core_number):
+            core_size.append(Size(1, Unit.GibiByte))
+        core_dev.create_partitions(core_size, partition_table)
+
+    with TestRun.step("Create filesystem on core devices."):
+        for i in range(core_number):
+            core_dev.partitions[i].create_filesystem(filesystem)
+
+    with TestRun.step("Mount core devices and create test files."):
+        counter = 1  # There are one partition (4th) which is omitted while creating cache
+        files = []
+        for core in core_dev.partitions:
+            mount_path = mount_point + str(counter)
+            core.mount(mount_path)
+            test_file_path = f"{mount_path}/test_file"
+            files.append(fs_utils.create_test_file(test_file_path))
+            counter += 1
+
+    with TestRun.step("Check md5 sums of test files."):
+        test_files_md5sums = []
+        for file in files:
+            test_files_md5sums.append(TestRun.executor.run(
+                f"md5sum {file.full_path}").stdout.split()[0])
+
+    with TestRun.step("Unmount core devices."):
+        for core in core_dev.partitions:
+            core.unmount()
+
+    with TestRun.step(f"Start cache."):
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+
+    with TestRun.step("Add cores to cache."):
+        cores = []
+        for i in range(core_number):
+            cores.append(cache.add_core(core_dev.partitions[i]))
+
+    with TestRun.step("Mount core devices."):
+        for core in cores:
+            mount_path = mount_point + str(core.core_id)
+            core.mount(mount_path)
+
+    with TestRun.step("Check again md5 sums of test files."):
+        test_files_md5sums_new = []
+        for file in files:
+            test_files_md5sums_new.append(TestRun.executor.run(
+                f"md5sum {file.full_path}").stdout.split()[0])
```

Comment:
1. Exit code should be checked when using TestRun.executor.run 
2. There is a 'md5sum' helper in our File class, please use that instead